url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/spec/args/ArgListBuilderUtil.java

```diff
@@ -0,0 +1,714 @@
+import com.sdicons.scripty.repl.ReplEngineException;
+import com.sdicons.scripty.spec.map.IArgMapping;
+import com.sdicons.scripty.spec.map.IndexedMapping;
+import com.sdicons.scripty.spec.map.PartialMapping;
+import com.sdicons.scripty.spec.type.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ArgListBuilderUtil
+{
+    private static ReplEngine typeProcessor = new ReplEngine();
+    
+    public static class TypeCommands
+    {
+        private static ITypeSpec SPEC_TYPE = new InstanceType(TypeTuple.class, false);
+        
+        private static TypeTuple INTEGER_TUPLE = new TypeTuple( new IntegerType(), null);
+        private static TypeTuple BIGDECIMAL_TUPLE = new TypeTuple(new BigDecimalType(), null);
+        private static TypeTuple BIGINTEGER_TUPLE = new TypeTuple(new BigIntegerType(), null);
+        private static TypeTuple BOOLEAN_TUPLE = new TypeTuple(new BooleanType(), null);
+        private static TypeTuple BYTE_TUPLE = new TypeTuple(new ByteType(), null);
+        private static TypeTuple DOUBLE_TUPLE = new TypeTuple(new DoubleType(), null);
+        private static TypeTuple FLOAT_TUPLE = new TypeTuple(new FloatType(), null);
+        private static TypeTuple LONG_TUPLE = new TypeTuple(new LongType(), null);
+        private static TypeTuple SHORT_TUPLE = new TypeTuple(new ShortType(), null);
+        private static TypeTuple STRING_TUPLE = new TypeTuple(new StringType(), null);
+        
+        private static NamedArg NAMED_MINLEN = new NamedArg("minLength", new IntegerRangeType(-1, Integer.MAX_VALUE), -1, true);
+        private static NamedArg NAMED_MAXLEN = new NamedArg("maxLength", new IntegerRangeType(-1, Integer.MAX_VALUE), -1, true);
+        private static NamedArg NAMED_MIN = new NamedArg("min", new IntegerRangeType(Integer.MIN_VALUE, Integer.MAX_VALUE), Integer.MIN_VALUE, true);
+        private static NamedArg NAMED_MAX = new NamedArg("max", new IntegerRangeType(Integer.MIN_VALUE, Integer.MAX_VALUE), Integer.MAX_VALUE, true);
+        private static NamedArg NAMED_NULLALLOWED = new NamedArg("nullAllowed", new BooleanType(), false, true);
+
+        private static FixedArg FIXED_TYPE = new FixedArg(SPEC_TYPE);
+        private static FixedArg FIXED_CLASSNAME = new FixedArg(new StringType());
+        private static VarArg VAR_STRING = new VarArg(new StringType());
+
+        private static StdArgList NOARGS = new StdArgList(new FixedArg[]{}, new OptionalArg[]{}, new NamedArg[]{});
+
+        @ScriptyCommand(name="BigDecimal")
+        public static TypeTuple bigDecimalType(Object[] aArgs, IContext aCtx)
+        throws CommandException
+        {
+            try
+            {
+                NOARGS.guard(aArgs, aCtx);
+                return BIGDECIMAL_TUPLE;
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 78:86, 93:101, 108:116, 123:131, 138:146, 185:193, 200:208, 314:322, 329:337, 344:352.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.